docs/java/how-to-set-activityoptions-in-java.md

```diff
@@ -0,0 +1,317 @@
+---
+id: how-to-set-activityoptions-in-java
+title: How to set ActivityOptions in Java
+sidebar_label: Activity Options
+description: Use `ActivityOptions` to configure how to invoke an Activity Execution.
+tags:
+  - java
+  - developer-guide
+---
+
+Use `ActivityOptions` to configure how to invoke an Activity Execution.
+Note that Activity options must be set before the Activity Execution is invoked.
+
+You can set Activity options for Activities within a Workflow or define specific Activity options per Activity Type within a Worker.
+
+## Setting Activity options within a Workflow
+
+Use `ActivityOptions` with `ActivityStub` to set options for invoking Activities within a Workflow.
+The following example shows how to set `ActivityOptions` for Activities within a Workflow.
+
+```java
+GreetingActivities activities = Workflow.newActivityStub(GreetingActivities.class,
+                ActivityOptions.newBuilder()
+                        .setStartToCloseTimeout(Duration.ofSeconds(5))
+                        // if task queue not set, it will be same Task Queue as what the Workflow uses
+                        .setTaskQueue("yourTaskQueue")
+                        // If RetryOptions are not explicitly set, Activities have a default RetryOption that apply.
+                       .setRetryOptions(RetryOptions.newBuilder()
+                                .build())
+                        .build());
+
+```
+
+Note that these Activity options will apply for all Activities within the Workflow.
+
+## Setting Ativity options within a Worker
```

Comment:
I would change this to "Setting per-activity options when registering a workflow with worker"

Reply:
done